C++/include/algorithm/dynamic_programming/cut_the_rod.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+    Cutting the rod problem
+    --------------------
+	Given a rod of length n inches and an array of prices 
+	that contains prices of all pieces of size smaller than n. 
+	Determine the maximum value obtainable by cutting up 
+	the rod and selling the pieces
+
+    Time complexity
+    ---------------
+    O(N^2)
+
+    Space complexity
+    ----------------
+    O(N)
+
+    Author
+    ------
+    Fedurko Nikolaus (@B1Z0N)
+*/
+
+#include <vector>	// vector
+
+#ifndef CUT_THE_ROD_HPP
+#define CUT_THE_ROD_HPP
+
+/**
+ * Recursive helper function
+ */
+int __cut_the_rod( const std::vector<int>& costs, std::vector<int>& visited, int len)
+{
+	if( len == 1 )
+		// whole unit, can't delete
+		return visited[0];
+
+	int cost = costs[ len - 1 ];
+	// try to sell it as is, not cutting
+	int next_cost;
+
+	for( int i = 1; i <= len / 2; i++ )
+	// loop to check only half
```

Comment:
Some of your comments are after the statement they reference, and some are before. Prefer consistency above all else, but I also think it would be better if the comments were before the code they reference.